case-app/case
# Readonly filters, persistent filters and log email

## What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)

This PR adds 3 new features:
- Persistent filters
- Readonly filters
- Log emails instead of sending

## Description

As said previously, adds 3 new features.

Note: For the log emails, I chose to console.log instead of logging into a file. The reason is that we do not use local log files in CASE but rather Bugsnag (and more monitoring tools in the future) for log management. I thought that it was overkill for logging emails, the main purpose of the feature is to prevent sending emails, not to monitor them.

## Related Issues

Not public

## How can it be tested?

- Download the new source code and install dependencies as always
- Persistent filters: Open the docs (run `npm run start:docs`) and check how to enable persistent filters
- Readonly filters: check how readonly filters work in the docs and then try on Select and MultiSelect inputs (only those 2 are developped yet). This feature also serves a "readonly" mode for fields in create-edit views. Just add `readonly: true` in a Select or MultiSelect field to make it reaonly.
- LOG Emails: Check how to do that the docs directly

## Does it causes a breaking change ?

YES. We have to pass the FilterService in the super() function in all list views.

## Check list before submitting
